pip/qsharp/interop/qasm/_ipython.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from ..._native import Output  # type: ignore
+
+_in_jupyter = False
+try:
+    from IPython.display import display
+
+    if get_ipython().__class__.__name__ == "ZMQInteractiveShell":  # type: ignore
+        _in_jupyter = True  # Jupyter notebook or qtconsole
+except:
+    pass
+
+
+def display_or_print(output: Output) -> None:
+    if _in_jupyter:
+        try:
+            display(output)
+            return
+        except:
+            # If IPython is not available, fall back to printing the output
+            pass
+    print(output, flush=True)
```

Comment:
Defining this is one place is nice... I wonder why we didn't do it for the code in pip/qsharp/_qsharp.py? If the helper logic in this file were moved to a more central place then everything could take advantage of it since this same pattern is used in three more places outside of your changes in existing code. Not critical for this change though.